src/lib/logger.ts

```diff
@@ -0,0 +1,61 @@
+import PrettyError from 'pretty-error'
+import * as process from 'process'
+import winston from 'winston'
+
+type LogLevel = 'error' | 'warn' | 'info' | 'debug'
+
+export type Logger = {
+  [method in LogLevel]: (message: string, info?: unknown) => void
+}
+
+const prettyError = new PrettyError()
+  .skipNodeFiles()
+  .skip(
+    (traceLine: Record<string, unknown>): boolean =>
+      traceLine.packageName !== '[current]'
+  )
+  .appendStyle({
+    'pretty-error': {
+      marginLeft: 0,
+    },
+    'pretty-error > trace': {
+      marginTop: 0,
+    },
+    'pretty-error > trace > item': {
+      bullet: '',
+      marginBottom: 0,
+    },
+  })
+
+const prettyErrorFormat = winston.format((info) => {
+  if (info.stack) {
+    return {
+      ...info,
+      message: prettyError.render(info),
+      stack: undefined,
+    }
+  }
+
+  return info
+})
+
+const format = winston.format.combine(
+  winston.format.colorize(),
+  winston.format.errors({ stack: true }),
+  prettyErrorFormat(),
+  winston.format.simple()
+)
+
+let logger: Logger = winston.createLogger({
+  level: process.env.LOG_LEVEL || 'info',
+  format,
+  transports: [new winston.transports.Console()],
+})
```

Comment:
For us it would be good if this logger logged in the ECS format (https://www.elastic.co/guide/en/ecs-logging/nodejs/current/winston.html) or if we could pass in another logger, otherwise the logs will appear as unparseable noise in kibana

Reply:
Ah okay I see that we can pass a logger, so maybe that's perfectly fine!